src/gui/TimePicker.ts

```diff
@@ -31,7 +31,7 @@ export class TimePicker implements Component<TimePickerAttrs> {
 				times.push(timeStringFromParts(hour, minute, attrs.amPmFormat))
 			}
 		}
-
+		this._oldValue = attrs.time?.toString(false) ?? "--"
```

Comment:
why is this necessary?

Reply:
having a not-null assertion on a field that's not assigned in the constructor is no bueno, changing all the call sites to decide what the default should be if they don't have an oldValue is also not great. 

just having the oldValue be something that can't be a time is easiest for the native time picker to not immediately call _onSelected on every first render.